tests/fixture_valid.js

```diff
@@ -246,14 +177,165 @@ module.exports.checkFixture = function checkFixture(fixture, usedShortNames=[])
             });
           }
 
-          if ('image' in cap && cap.image.length > 0 && 'color' in cap) {
+          if ('color' in cap && 'image' in cap) {
             result.errors.push({
               description: `color and image cannot be present at the same time in capability #${i} in channel '${ch}'.`,
               error: null
             });
           }
+
+          if ('switchToChannels' in cap) {
+            if (switchesCount === 0) {
+              result.errors.push({
+                description: `Capability '${cap.name}' (#${i}) uses the 'switchToChannels' property, but its channel '${ch}' is missing the 'switchesChannels' property.`,
+                error: null
+              });
+            }
+            else if (cap.switchToChannels.length !== switchesCount) {
+              result.errors.push({
+                description: `Capability '${cap.name}' (#${i}) uses ${cap.switchToChannels.length} channel(s) in 'switchToChannels', but its channel '${ch}' uses ${switchesCount} channel(s) in 'switchesChannels'.`,
+                error: null
+              });
+            }
+
+            // switched channels are used with the switching channel and don't need to be used somewhere else
+            for (const switchedChannel of cap.switchToChannels) {
+              if (!(switchedChannel in fixture.availableChannels)) {
```

Comment:
I didn't allow switch fine channels by purpose as I don't think that it makes much sense and it only makes things even more complicated. What do you think about it?

However, we can still add the functionality as soon as we see a fixture that switches fine channels.